app/controllers/application_controller.rb

```diff
@@ -45,6 +45,42 @@ def self.after_update_path_for(resource)
 
   private
 
+  def download_or_show_image(type, render_to, width)
+    html = image_html(type)
+
+    render html: html.html_safe and return unless render_to == 'jpg'
+
+    kit = build_kit(html, "#{type.tr('_', '-')}.css", width)
+
+    send_data(kit.to_jpg, type: 'image/jpg', filename: "#{type}.jpeg")
+  end
+
+  def image_html(image_type)
+    render_to_string(partial: image_type,
+                     locals: { app_config: @app_configuration, user: @user,
+                               render_to: params[:render_to]&.to_sym,
+                               context: params[:context]&.to_sym,
+                               company: Company.find_by_id(params[:company_id]) })
+  end
+
+  def build_kit(html, image_css, width)
+    kit = IMGKit.new(html, encoding: 'UTF-8', width: width, quality: 100)
+    kit.stylesheets << Rails.root.join('app', 'assets', 'stylesheets',
+                                       image_css)
+    kit
+  end
+
+  def set_app_config
+    # Need app config items for proof-of-membership
+    @app_configuration = AdminOnly::AppConfiguration.last
+  end
+
+  def allow_iframe_request
+    response.headers.delete('X-Frame-Options')
+    # https://stackoverflow.com/questions/17542511/
+    # cannot-display-my-rails-4-app-in-iframe-even-if-x-frame-options-is-allowall
+  end
```

Comment:
All of these files should _not_ be added to the application controller file - that should be limited to methods that are common to all (or most) controllers.   That is not the case with these methods.

I know that this is a way to put these methods into the scope of the two controllers (companies, users), but the Rails way of doing that is to use a common "concerns" file, that is included in the two controllers.

For an example, see `controllers/concerns/pagination_utility.rb`.   That file is included in any controller that sets up JS-driven pagination for tables in views.

Really, what you have to do is define a module - which contains the shared methods - and include the statement `extend ActiveSupport::Concern` in that.  The methods in the module file will be instance methods in the including class (e.g, for the controller instance).

This provides a good overview of the pattern that `ActiveSupport::Concern` is enforcing: http://engineering.appfolio.com/appfolio-engineering/2013/06/17/ruby-mixins-activesupportconcern

Reply:
I added `extend ActiveSupport::Concern` but I later did cucumber tests for `features/user_account/*` without it and it works okay. I don't think that there's used any of `ActionController::Base` methods (there are `render_to_string` from `ActionController::Rendering` and `send_data` from `ActionController::DataStreaming`) so is it really necessary? 

Also, does it work without `included` method like in `pagination_utility.rb`?

Reply:
@tmszrbn - you're right .... given that we're just adding instance methods we don't need the `ActiveSupport::Concern` module to be included.

And, yes, you don't need the `included` method since we're not taking any special action when the module is actually included in the controller class.